docs/en/cookbook/dql-user-defined-functions.rst

```diff
@@ -3,69 +3,63 @@ DQL User Defined Functions
 
 .. sectionauthor:: Benjamin Eberlei <kontakt@beberlei.de>
 
-By default DQL supports a limited subset of all the vendor-specific
-SQL functions common between all the vendors. However in many cases
-once you have decided on a specific database vendor, you will never
-change it during the life of your project. This decision for a
-specific vendor potentially allows you to make use of powerful SQL
-features that are unique to the vendor.
```

Comment:
Why are you removing these 2 sentences? I think they are very interesting.

Reply:
For 2 reasons:
* It's an assumption. Maybe the user changes the database vendor every week...
* It's not relevant here. People who come here want to create a user-defined function. The sooner the page gets to that point, the better.

Reply:
> However in many cases

It leaves room for the unlikely case where the user changes the database vendor every week

> It's not relevant here. People who come here want to create a user-defined function. The sooner the page gets to that point, the better.

I think it's nice to know that it might be completely fine to write user-defined-function, and why they are not built into Doctrine. I do agree that it's kind of secondary to this how to, so maybe it could become a note?

Reply:
* "in many cases" sounds patronizing. If I buy a toaster, I don't want to read that people usually eat toasts for breakfast ;-)
* Completely fine: Sure! But that's obvious anyway! Otherwise there wouldn't be an article in the docs about it. And even the alternative (raw SQL query) is explicitly mentioned.